src/main/java/org/folio/reshare/index/storage/Storage.java

```diff
@@ -50,23 +61,62 @@ public Storage(Vertx vertx, String tenant) {
                 + "(id uuid NOT NULL PRIMARY KEY,"
                 + "local_identifier VARCHAR NOT NULL,"
                 + "library_id uuid NOT NULL,"
-                + "title VARCHAR,"
-                + "match_key VARCHAR NOT NULL,"
-                + "isbn VARCHAR,"
-                + "issn VARCHAR,"
-                + "publisher_distributor_number VARCHAR,"
                 + "source JSONB NOT NULL,"
                 + "inventory JSONB"
                 + ")",
             "CREATE UNIQUE INDEX IF NOT EXISTS idx_local_id ON " + bibRecordTable
                 + " (local_identifier, library_id)",
-            "CREATE INDEX IF NOT EXISTS idx_bib_record_match_key ON " + bibRecordTable
-                + " (match_key)",
             "CREATE OR REPLACE VIEW " + itemView
-                + " AS SELECT id, local_identifier, library_id, match_key,"
+                + " AS SELECT id, local_identifier, library_id,"
                 + " jsonb_array_elements("
                 + " (jsonb_array_elements((inventory->>'holdingsRecords')::JSONB)->>'items')::JSONB"
-                + ") item FROM " + bibRecordTable
+                + ") item FROM " + bibRecordTable,
+            "CREATE TABLE IF NOT EXISTS " + matchTypeTable
+                + "(id uuid NOT NULL PRIMARY KEY,"
+                + " code VARCHAR, "
+                + " name VARCHAR)",
+            "CREATE UNIQUE INDEX IF NOT EXISTS idx_match_type_code ON " + matchTypeTable
+                + " (code)",
+            "INSERT INTO " + matchTypeTable
+                + " (id, code, name) "
+                + " VALUES ('" + MATCH_TYPE_ID_ISBN + "', 'ISBN', 'ISBN') ON CONFLICT DO NOTHING",
+            "CREATE TABLE IF NOT EXISTS " + matchPointTable
+                + "(bib_record_id uuid NOT NULL,"
+                + " match_type_id uuid NOT NULL,"
+                + " match_value VARCHAR NOT NULL,"
+                + " CONSTRAINT match_point_fk_bib_record FOREIGN KEY "
+                + "                (bib_record_id) REFERENCES " + bibRecordTable
+                + " ON DELETE CASCADE,"
+                + " CONSTRAINT match_point_fk_type FOREIGN KEY "
+                + "                (match_type_id) REFERENCES " + matchTypeTable + ")",
+            "CREATE UNIQUE INDEX IF NOT EXISTS match_point_idx ON " + matchPointTable
+                + " (match_type_id, match_value, bib_record_id)",
+            "CREATE INDEX IF NOT EXISTS match_point_bib_id_idX ON " + matchPointTable
+                + " (bib_record_id)",
+            "CREATE OR REPLACE FUNCTION " + matchPointFunction
```

Comment:
Let's avoid hard-coding any specific match key methods here. These should be "pluggable" so let's factor them out to a separate class or classes.